common/lib/xmodule/xmodule/xml_module.py

```diff
@@ -428,6 +429,13 @@ def add_xml_to_node(self, node):
         """
         # Get the definition
         xml_object = self.definition_to_xml(self.runtime.export_fs)
+        for aside in self.runtime.get_asides(self):
+            if aside.needs_serialization():
+                aside_node = etree.Element("unknown_root", nsmap=XML_NAMESPACES)
+                aside.has_children = False
```

Comment:
That's a bug in xblock that it would be great if you could fix. `HierarchyMixin` and `XmlSerializationMixin` are independent (as you gathered), so we shouldn't assume the former in the code for the latter.